### ISSUE TITLE
PDF Image Not Being Rendered

### ISSUE THREAD

**Issue by srshupe** [mozilla/pdf.js#12307]
Attach (recommended) or Link to PDF file here: https://drive.google.com/file/d/1fwumi5lTjjA1enmeQVSC-tDJAZLL3pyI/view

Configuration:
- Web browser and its version: 
Chrome Version 85.0.4183.83 
Firefox 80.0 (64-bit)
(worked in Microsoft Edge 44.17763.831.0)
- Operating system and its version:
Windows 10 v1809 build 17763.1397
- PDF.js version: 2.2.228, also tested using viewer at https://mozilla.github.io/pdf.js/web/viewer.html (2.5.207?)
- Is a browser extension: no

Steps to reproduce the problem:
1. Open the linked PDF in Adobe Reader or Acrobat, or a browser. The images are rendered.
2. Open the linked PDF in pdf.js using Chrome or Firefox. Most or all of the images are not rendered.

What is the expected behavior? (add screenshot)
For the images in the PDF to be displayed in pdf.js for all browsers:
https://drive.google.com/file/d/15_9mRv2p-1c3AUOGKIJ7w2u2heV4Mt4M/view

What went wrong? (add screenshot)
The images were not displayed in Chrome or Firefox.
In Chrome, some parts of the PDF was rendered and there were no reported issues:
https://drive.google.com/file/d/1REE9qOP5UQirDs-ZKEhcWykvWY3w8ORv/view

In Firefox an error was reported in the UI:
https://drive.google.com/file/d/1nLl4Rd3aq9DYtTOc3seRWLsWbP1nbSJL/view

In Microsoft Edge using pdfj.js v2.2.228 the complete PDF was displayed. The viewer at https://mozilla.github.io/pdf.js/web/viewer.html didn't work with Edge.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):
https://mozilla.github.io/pdf.js/web/viewer.html

**Comment by SamyCookie** [comment]
Here are the exceptions got during rendering (with Firefox 88):
```
PDF 110dd61fd57444010b1ab5ff38782f0f [1.4 pdfeTeX-1.21a / TeX] (PDF.js: 2.9.273) app.js:1516:4
PDF c10b27792222dd4fcc446d0bc858d91a [1.7 - / -] (PDF.js: 2.9.273) app.js:1516:4
renderView: "[Exception... "Failure"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: https://mozilla.github.io/pdf.js/build/pdf.js :: putBinaryImageData :: line 4938"  data: no]" pdf_rendering_queue.js:178:12
    renderView pdf_rendering_queue.js:178
    (Asynchrone : FrameRequestCallback)
    _scheduleNext api.js:3152
    renderContinueCallback pdf_page_view.js:516
    onContinue pdf_page_view.js:735
    _continue api.js:3144
    operatorListChanged api.js:3120
    _renderPageChunk api.js:1629
    pump api.js:1666
renderView: "[Exception... "Failure"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: https://mozilla.github.io/pdf.js/build/pdf.js :: putBinaryImageData :: line 4938"  data: no]" pdf_rendering_queue.js:178:12
    renderView pdf_rendering_queue.js:178
    (Asynchrone : promise callback)
    renderView pdf_rendering_queue.js:169
    forceRendering base_viewer.js:1241
    (Asynchrone : promise callback)
    forceRendering base_viewer.js:1240
    renderHighestPriority pdf_rendering_queue.js:73
    renderView pdf_rendering_queue.js:171
    _scheduleNext api.js:3152
    renderContinueCallback pdf_page_view.js:516
    onContinue pdf_page_view.js:735
    _continue api.js:3144
    operatorListChanged api.js:3120
    _renderPageChunk api.js:1629
    pump api.js:1666
```

**Comment by casper5822** [comment]
We have the same problem with advanced and recent mobile smartphones.

For example users with Samsung Galaxy S21 Plus don't display images in some pdf page.

We tested it on Firefox and Chrome.

Pdf.js version: 2.13.216
OS: Android 11
Web app: vanillajs

In attachment the user screenshot on galaxy s21 plus.

![photo5816553207315414013](https://user-images.githubusercontent.com/26361006/172114497-12122dcb-cb7a-4a57-8744-9e1282a7f915.jpg)

**Comment by blackmambahk** [comment]
Is this issue being worked on?
Is there a likelihood of it being resolved this year?
Is there currently some suggested workaround? 
Thanks

**Comment by prmahima** [comment]
It would be helpful if anyone could provide an update on this. As asked earlier in previous comment-
Is this issue being worked on?
Is there a likelihood of it being resolved this year?
Is there currently some suggested workaround?
Thanks

**Comment by Snuffleupagus** [comment]
Fixed by PR #16077.

**Comment by VaibhavMalokar** [comment]
Hello guys,
Just want to know if this release is available in the latest build (version 3.5.141) or not. Thanks.

**Comment by dhanbhan** [comment]
@Snuffleupagus I see this issue (#14800 ) is closed on April 2022. I have verified in latest pdfjs build `pdfjs-3.6.172` with same test document provided, I see blank page for 15-20 seconds and page displays at later.

May I know if it is already delivered in `pdfjs-3.6.172` ? If yes I don't see it is working as expected.
